charms/modeldb-backend/reactive/modeldb_backend.py

```diff
@@ -55,7 +56,36 @@ def start_charm():
                         {
                             'name': 'config',
                             'mountPath': '/config-backend/',
-                            'files': {'config.yaml': hookenv.config('config')},
+                            'files': {
+                                'config.yaml': yaml.dump(
+                                    {
+                                        'grpcServer': {'host': hookenv.service_name(), 'port': grpc_port},
+                                        'database': {
+                                            'DBType': 'rdbms',
+                                            'RdbConfiguration': {
+                                                'RdbDatabaseName': 'mariadb',
+                                                'RdbDriver': 'com.mysql.cj.jdbc.Driver',
+                                                'RdbDialect': 'org.hibernate.dialect.MySQL5Dialect',
+                                                'RdbUrl': f'jdbc:mysql://{mysql.host()}:{mysql.port()}',
+                                                'RdbUsername': 'root',
+                                                'RdbPassword': hookenv.config('mysql-root-password'),
```

Comment:
This should come from the relation as well, since it has to match what's on the remote side.